src/FileCache.php

```diff
@@ -343,7 +346,17 @@ private function normalizeTtl(null|int|string|DateInterval $ttl = null): ?int
      */
     private function createDirectoryIfNotExists(string $path): bool
     {
-        return is_dir($path) || (!is_file($path) && mkdir($path, $this->directoryMode, true) && is_dir($path));
+        if (true === is_dir($path)) {
+            return true;
+        }
+
+        $result = !is_file($path) && mkdir(directory: $path, recursive: true) && is_dir($path);
```

Comment:
Any reason not to use [mkdir's permissions parameter](https://www.php.net/manual/en/function.mkdir.php)?